versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5804 @@
+```wdl
+Boolean morning = ...
+String greeting = "good " + if morning then "morning" else "afternoon"
+```
+- Choose how much memory to use for a task:
+```wdl
+Int array_length = length(array)
+runtime {
+  memory: if array_length > 100 then "16GB" else "8GB"
+}
+```
+
+#### Function Calls
+
+WDL provides a [standard library](#standard-library) of functions. These functions can be called using the syntax `func(p1, p2, ...)`, where `func` is the function name and `p1` and `p2` are parameters to the function. Note that standard library function names are identifiers, meaning they are reserved for use by WDL and cannot be used as declaration, task, or workflow names or aliases.
+
+#### Expression Placeholders and String Interpolation
+
+Any WDL string expression may contain one or more "placeholders" of the form `~{*expression*}`, each of which contains a single expression. Note that placeholders of the form `${*expression*}` may also be used interchangably, but their use is discouraged for reasons discussed in the [command section](#expression-placeholders) and may be deprecated in a future version of the specification.
+
+When a string expression is evaluated, its placeholders are evaluated first, and their values are then substituted for the placeholders in the containing string.
+
+```wdl
+Int i = 3
+String s = "~{1 + i}"  # s == "4"
+```
+
+As another example, consider how the following expression would be parsed:
+
+```wdl
+String command = "grep '~{start}...~{end}' ~{input}"
+```
+
+This command would be parsed as:
+
+* `grep '` - literal string
+* `~{start}` - identifier expression - replaced with the value of `start`
+* `...` - literal string
+* `~{end}` - identifier expression - replaced with the value of `end`
+* `' ` - literal string
+* `~{input}` - identifier expression - replaced with the value of `input`
+
+Placeholders may contain other placeholders to any level of nesting, and placeholders are evaluated recursively in a depth-first manner. Placeholder expressions are anonymous - they have no name and thus cannot be referenced by other expressions, but they can reference declarations and call outputs.
+
+```wdl
+Int i = 3
+Boolean b = true
+# s evaluates to "4", but would be "0" if b were false
+String s = "~{if b then '${1 + 3}' else 0}"
+```
+
+##### Expression Placeholder Coercion
+
+The expression result in a placeholder must ultimately be converted to a string in order to take the place of the placeholder in the command script. This is immediately possible for WDL primitive types due to automatic conversions ("coercions") that occur only within the context of string interpolation:
+
+- `String` is substituted directly.
+- `File` is substituted as if it were a `String`.
+- `Int` is formatted without leading zeros (unless the value is `0`), and with a leading `-` if the value is negative.
+- `Float` is printed in the style `[-]ddd.ddd`, with 6 digits after the decimal point.
```

Comment:
Excluding the decimal point?  or did you mean `[-]ddd.dddddd`?

Reply:
yep good catch